utils/set/set.go

```diff
@@ -192,3 +192,17 @@ func (s *Set[T]) Peek() (T, bool) {
 	}
 	return utils.Zero[T](), false
 }
+
+// Intersect returns the set intersection of s1 and s2
+func Intersect[T comparable](s1 Set[T], s2 Set[T]) Set[T] {
```

Comment:
I thought this looked better than putting it on a receiver... but admittedly it is inconsistent w/ the rest of the code.